### ISSUE TITLE
create a release.yml file for fabric8 projects to describe what steps are needed for its release

### ISSUE THREAD

**Issue by rawlingsj** [fabric8io/fabric8#5348]
TL;DR move this out of the workflow..
https://github.com/fabric8io/jenkins-workflow-library/blob/master/vars/releaseProject.groovy#L13-L33

Currently the fabric8 release workflow scripts contain the information about each project of how to release it.  I.e. it contains a list of docker images to tag in docker hub, images to promote, dependencies to update and whether to generate helm charts and push to the helm repository.

This logic should be moved out of the workflow scripts so that each project is responsible for describing the release tasks.  The workflow scripts can then read the release.yml (or something similar) to figure out what steps / workflow functions are needed.
